source/MaterialXView/Viewer.cpp

```diff
@@ -810,6 +806,17 @@ void Viewer::loadDocument(const mx::FilePath& filename, mx::DocumentPtr librarie
         for (const auto& renderablePath : renderablePaths)
         {
             mx::ElementPtr elem = doc->getDescendant(renderablePath);
+            if (!elem->getAttribute("node").empty())
```

Comment:
There are other situations in which we'll need to update the "shading model" display for the current material, e.g. when the user changes the selected material or selected geometry.  To keep everything in sync, I would recommend extending the existing Viewer::updatePropertyEditor method, making it general enough to include your new functionality as well.  One potential name for the new, more comprehensive method would be Viewer::updateDisplayedProperties.

Reply:
Sounds good! At the moment, each time a new material is chosen, loadDocument is called and then loadDocument calls updateDisplayedProperties. Do you suggest I move the updateDisplayedProperties outside of loadDocument?

Reply:
My thinking is that we would replace updatePropertyEditor with the more general updateDisplayedProperties, and then call the new method from all of the old locations where we used to call updatePropertyEditor.  That would include loadDocument, but also other locations where the user is switching between materials or geometries.